src/ArrowFunctions.cpp

```diff
@@ -849,6 +876,24 @@ int cpp_readColumnByName(const char* filename, void* chpl_arr, const char* colna
           delete[] def_lvl;
           delete[] rep_lvl;
         }
+      } else if(ty == ARROWDECIMAL) {
+        auto chpl_ptr = (double*)chpl_arr;
+        parquet::FixedLenByteArray value;
+        parquet::FixedLenByteArrayReader* reader =
+          static_cast<parquet::FixedLenByteArrayReader*>(column_reader.get());
+        startIdx -= reader->Skip(startIdx);
+
+        while (reader->HasNext() && i < numElems) {
+          if((numElems - i) < batchSize)
+            batchSize = numElems - i;
+          (void)reader->ReadBatch(1, nullptr, nullptr, &value, &values_read);
```

Comment:
I don't really see how `batchSize` factors into this since we are always calling `ReadBatch` with a size of 1

Reply:
You are right, that isn't doing anything in this case, it was just leftover from copy/paste of the integer section. I'll remove that.